File: suwmp_fe/src/types/enterpriseCapacity.ts
export type CapacityStatus = "NORMAL" | "WARNING" | "CRITICAL";

export interface WasteType {
  id: number;
  name: string;
}

export interface Enterprise {
  id: number;
  name: string;
}

export interface EnterpriseCapacity {
  id: number;
  wasteTypeId: number;
  wasteTypeName: string;
  enterpriseId: number;
  dailyCapacityKg: number;
  warningThreshold: number;
  active: boolean;
  usedKg: number;
}

export interface UpdateEnterpriseCapacityRequest {
  dailyCapacityKg: number;
  warningThreshold: number;
  active: boolean;
}

export interface CreateEnterpriseCapacityRequest {
  enterpriseId: number | null;
  wasteTypeId: number;
  dailyCapacityKg: number;
  warningThreshold: number;
}

File: suwmp_be/src/main/java/com/example/suwmp_be/entity/EnterpriseCapacity.java
package com.example.suwmp_be.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "enterprise_capacity")
public class EnterpriseCapacity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "enterprise_id", nullable = false)
    @JsonIgnore
    private Enterprise enterprise;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "waste_type_id", nullable = false)
    @JsonIgnore
    private WasteType wasteType;

    @Column(nullable = false)
    private int dailyCapacityKg;

    @Column(nullable = false)
    private int warningThreshold;

    @Column(nullable = false)
    private boolean active;
}
